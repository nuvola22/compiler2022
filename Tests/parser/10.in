procedure a();
begin
    writeln('Hello, world');
end;

procedure b(var c: Integer; const e: String);
begin
    writeln(e, c);
end;

procedure d();
const hello = 'Hello, world';
begin
    writeln(hello);
end;

begin end.