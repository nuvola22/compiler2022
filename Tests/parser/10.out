      program
         a
            parameters:
            declarations:
            begin
               call
                  writeln
                  'Hello, world'
         b
            parameters:
               Var
               c
               Integer
               Const
               e
               String
            declarations:
            begin
               call
                  writeln
                  e
                  c
         d
            parameters:
            declarations:
               constant
                  hello
                  'Hello, world'
            begin
               call
                  writeln
                  hello
         begin
